manifests/server/config.pp

```diff
@@ -154,17 +154,23 @@
   }
 
   # autosign file
-  if $::puppet::server_ca and ! is_bool($puppet::autosign) {
-    file { $puppet::autosign:
-      ensure => file,
-      owner  => $puppet::user,
-      group  => $puppet::group,
-      mode   => $puppet::autosign_mode,
-    }
-    if !empty($puppet::autosign_entries) {
-      File[$puppet::autosign] {
-        content => template('puppet/server/autosign.conf.erb'),
+  if $::puppet::server_ca and ! is_bool($puppet::server::autosign){
+    if $::puppet::server::autosign_content {
+      if !empty($::puppet::server::autosign_entries) {
+        fail('Cannot set both autosign_content and autosign_entries')
```

Comment:
could you add a test for this? grep for `it { should raise_error` in the existing tests.